### PR TITLE
Rename can_be_instructor and can_be_participant to instructor and participant

### PR DESCRIPTION
**Pull request by dmcavoy**
When I was talking with @bencodeorg about instructor and participant audience stuff I noticed that the names of these methods was not capturing that these methods can tell you if someone is the instructor or participant for a course. So I updated their names to better capture that.

## Links

[Jira](https://codedotorg.atlassian.net/browse/PLAT-1732)

## Testing story

- Tests updated too

### CODE REVIEW

File: dashboard/app/models/concerns/curriculum/course_types.rb [hunk 1]
```diff
@@ -54,11 +54,11 @@ def cannot_have_same_audiences
   # Checks if a user can be the instructor for the course. universal instructors and levelbuilders
   # can be the instructors of any course. Student accounts should never be able to be the instructor
   # of any course.
-  def can_be_instructor?(user)
+  def instructor?(user)
```
Comment (davidsbailey) [inline dashboard/app/models/concerns/curriculum/course_types.rb:57]: I am a little confused. the comment says "Checks if a user **can be** the instructor for the course". are we checking whether the user **is** or **can be**?

**Comment by bencodeorg** [comment]
This is maybe out of scope for this PR, but if we're thinking about how to make these easier to understand, I feel like this would be easier to read if these methods were defined on the user model (or included there) like `user.can_be_instructor(script)`:

https://github.com/code-dot-org/code-dot-org/blob/5af2e7f7286cde5c349d68ff30c1fe80f2b04103/dashboard/app/models/ability.rb#L281-L293

Sorry if this is a totally misinformed idea...

**Comment by davidsbailey** [comment]
> This is maybe out of scope for this PR, but if we're thinking about how to make these easier to understand, I feel like this would be easier to read if these methods were defined on the user model (or included there) like `user.can_be_instructor(script)`:

hey Ben, this is a good thought, since it makes readability a top priority, and may be the right call in other cases. but for the user model in particular, our user.rb file is so big that we try to avoid adding new things to it. 

there is a whole other argument that methods which compute business logic really should be "policy" objects (they can live in dashboard/lib/policies) and be called like `CourseInstructorPolicy.can_user_instruct?(user: u, course: c)`. we're a long way off from that, but it is one direction we might move in if / when we try to split up the user model.

File: dashboard/app/models/concerns/curriculum/course_types.rb (hunk 1 above)
Comment (dmcavoy) [reply dashboard/app/models/concerns/curriculum/course_types.rb:57]: I will update the comment.

**Comment by dmcavoy** [comment]
@bencodeorg I definitely agree it would be easier to read if it was on the user model. We consider it at the beginning of this project and in addition to what @davidsbailey said there is also the complicating factor of the fact that we have to do different things when the top level of a course is a UnitGroup vs a Script so it also was easier to implement this way because of that. Hopefully some of that messiness will be getting better soon.

**Comment by bencodeorg** [comment]
Thanks for the explanation! One other idea while I'm here lobbing partially informed opinions, could there be a `role(user)` helper that makes explicit that a user can either be an instructor or a participant (or maybe third category is unauthorized or something?) for a given course (and returns that role)? I see there's a comment in the `participant?` method to this effect, but might be a readability upgrade for the future.

File: dashboard/app/models/concerns/curriculum/course_types.rb (hunk 1 above)
Comment (davidsbailey) [reply dashboard/app/models/concerns/curriculum/course_types.rb:57]: ok, I see that the comment is consistent now, but I think I am missing the bigger picture here. 

to me, `unit_group.can_be_instructor?(user)` says "can this user be an instructor of this unit group?", which sounds accurate based on what the code is doing.

by contrast, `unit_group.instructor?(user)` sounds like "is the user the instructor of this unit group?", which does not match my understanding of what the code is doing.

is there something I am missing?

Comment (dmcavoy) [reply dashboard/app/models/concerns/curriculum/course_types.rb:57]: I think the issue that was not clear before that this methods definitively tell you if someone is the instructor or the participant. They can not be be both. 

It sounded like that might not have been clear before. I'm happy to leave them as they are. Also I tried to use `is_instructor?` and `is_participant?` but our style checker did not like that

Comment (davidsbailey) [reply dashboard/app/models/concerns/curriculum/course_types.rb:57]: "is the instructor" meaning that they currently own a section which is assigned to this unit group? that's where I'm getting tripped up.

Comment (davidsbailey) [reply dashboard/app/models/concerns/curriculum/course_types.rb:57]: to me, "can be instructor" sounds like the exactly what I understand the code to be doing -- is the user allowed to assign a section they own to this unit group.

I'm trying not to make my opinion sound like fact, but I am trying to figure out where our understandings diverge as a first step to reaching an understanding.

**Comment by dmcavoy** [comment]
It sounds like this is not a big enough improvement or possibly not an improvement at all so I'm going to close this PR. Thanks for the feedback

**Pull request closed without merging**